### PR TITLE
Handle screen rotation for Leo panel

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/35806

This PR adds `android:configChanges="orientation|screenSize"` attributes to the manifest for `BraveLeoActivity` to prevent Activity destroy and re-creation during screen orientation change, [reference](https://developer.android.com/reference/android/app/Activity#ConfigurationChanges).

https://github.com/brave/brave-core/assets/24739341/95f6052d-00d9-4096-8eb4-5fa7afe1c270

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Enable screen rotation on the device
2. Launch browser, open Leo panel
3. rotate the screen

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java [hunk 1]
```diff
@@ -330,6 +331,12 @@ protected void onOrientationChange(int newOrientation) {
             setBottomToolbarVisible(isBottomToolbarVisible);
         }
 
+        if (mActivity instanceof BraveLeoActivity) {
+            // When Leo panel is shown on rotated screen we don't care about
+            // the toolbar.
+            return;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java:337]: is there still bottom toolbar when you close Leo Activity after a rotation?

Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java:337]: No, when I close Leo Activity after a rotation, there is no bottom toolbar. 

But the same behavior is on our `Nightly` and `Stable`, there is no bottom toolbar in landscape orientation  even without Leo.


https://github.com/brave/brave-core/assets/24739341/17b95c0d-bc8f-4493-9eb4-6d0cdbd67333

Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java:337]: ah yeah, we turn it off on landscape mode, yes, we are good in that case.

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java (hunk 1 above)
Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java:337]: > I wonder if bottom toolbar is scrolled properly on page scrolling after Leo activity closed as we don't apply adjustment in this case.

@samartnik yes, the bottom toolbar get hidden while scrolling after Leo activity closed, as I understand


https://github.com/brave/brave-core/assets/24739341/4e2f3f9b-1221-48f8-8060-6ecc0e3f162f

Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/toolbar/BraveToolbarManager.java:337]: here how does it work with this plan
1. open several tabs in group to have double toolbar
2. switch to landscape orientation
3. open Leo
4. switch back to portrait orientation and close Leo
5. check weather during scroll the toolbar will be hidden in a normal way


https://github.com/brave/brave-core/assets/24739341/11346199-c61c-4f27-ad0d-a7b9e1a3737d

**Pull request merged**